addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -1464,6 +1464,231 @@ QUnit.test('chat window: switch on TAB', async function (assert) {
     );
 });
 
+QUnit.test('chat window: cycle between more than 2 chat windows (LTR)', async function (assert) {
+    /**
+     * computation uses following info:
+     * ([mocked] global window width: @see `mail/static/src/utils/test_utils.js:start()` method)
+     * (others: @see mail/static/src/models/chat_window_manager/chat_window_manager.js:visual)
+     *
+     * - chat window width: 325px
+     * - start/end/between gap width: 10px/10px/5px
+     * - hidden menu width: 200px
+     * - global width: 1920px
+     *
+     * Enough space for 2 visible chat windows:
+     *  10 + 325 + 5 + 325 + 10 = 670 < 1920
+     */
+    assert.expect(6);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 1,
+                is_minimized: true,
+                is_pinned: true,
+                name: "channel1",
+                state: 'open',
+            }, {
+                channel_type: "channel",
+                id: 2,
+                is_minimized: true,
+                is_pinned: true,
+                name: "channel2",
+                state: 'open',
+            }, {
+                channel_type: "channel",
+                id: 3,
+                is_minimized: true,
+                is_pinned: true,
+                name: "channel3",
+                state: 'open',
+            }],
+        },
+    });
+    await this.start({
+        env: {
+            browser: {
+                innerWidth: 1920,
+            },
+        },
+    });
+    assert.containsN(
+        document.body,
+        '.o_ChatWindow .o_ComposerTextInput_textarea',
+        3,
+        "initialy, 3 chat windows should be present"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ChatWindow.o-folded',
+        "all 3 chat windows should be open"
+    );
+
+    const chatWindows = document.querySelectorAll('.o_ChatWindow');
+    await afterNextRender(() => {
+        chatWindows[2].querySelector('.o_ComposerTextInput_textarea').focus();
```

Comment:
It would be best if we could find another way (than using index) to identify chat windows based on their position, since their order in the DOM might be different than their actual "style" position.

This makes the correctness of the test hard to validate by just reading the code, which also makes its maintenance hard.

cc @alexkuhn

Reply:
Agree relying on index of `document.querySelectorAll()` is not nice. Enforcing n-th in DOM = n-th chat window, regardless of LTR/RTL is bad, so that's not a good alternative neither.

I propose using a `data-order` (just a demo name) on `ChatWindow` that tell its order. That way, tests have to sort `document.querySelectorAll()` using this `data-order`, so that order of chat window is more meaningful.

If we really really want to check visual placement of the chat windows, we should use something more powerful like `.getBoundingClientRect()`.